kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/ColumnMappingSuite.scala

```diff
@@ -257,284 +251,280 @@ class ColumnMappingSuite extends AnyFunSuite {
-      .hasSize(2)
-      .anySatisfy((k: AnyRef, v: AnyRef) => {
-        assertThat(k).asString.startsWith("col-")
-        assertThat(k).asString.endsWith(".key")
-        assertThat(v).isEqualTo(9L)
-      })
-      .anySatisfy((k: AnyRef, v: AnyRef) => {
-        assertThat(k).asString.startsWith("col-")
-        assertThat(k).asString.endsWith(".value")
-        assertThat(v).isEqualTo(10L)
-      })
-
-    assertThat(mapType.getKeyField.getMetadata.getEntries)
-      .doesNotContainKey(ColumnMapping.COLUMN_MAPPING_ID_KEY)
-      .doesNotContainKey(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY)
-
-    assertThat(mapType.getValueField.getMetadata.getEntries)
-      .doesNotContainKey(ColumnMapping.COLUMN_MAPPING_ID_KEY)
-      .doesNotContainKey(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY)
-
-    // verify nested ids
-    assertThat(innerStruct.get("f").getMetadata.getEntries
-      .get(COLUMN_MAPPING_NESTED_IDS_KEY).asInstanceOf[FieldMetadata].getEntries)
-      .hasSize(1)
-      .anySatisfy((k: AnyRef, v: AnyRef) => {
-        assertThat(k).asString.startsWith("col-")
-        assertThat(k).asString.endsWith(".element")
-        assertThat(v).isEqualTo(11L)
-      })
-
-    assertThat(metadata.getConfiguration)
-      .containsEntry(ColumnMapping.COLUMN_MAPPING_MAX_COLUMN_ID_KEY, "11")
-  }
+        true).get()
 
-  private def createMetadata(schema: StructType) = new Metadata(
-    UUID.randomUUID.toString,
-    Optional.empty(),
-    Optional.empty(),
-    new Format,
-    schema.toJson,
-    schema,
-    VectorUtils.buildArrayValue(Collections.emptyList(), StringType.STRING),
-    Optional.empty(),
-    VectorUtils.stringStringMapValue(Collections.emptyMap()))
-
-  private def assertColumnMapping(
-      field: StructField,
-      expId: Long,
-      expPhysicalName: String = "UUID"): Unit = {
-    assertThat(field.getMetadata.getEntries)
-      .containsEntry(ColumnMapping.COLUMN_MAPPING_ID_KEY, expId.asInstanceOf[AnyRef])
-      .hasEntrySatisfying(
-        ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY,
-        (k: AnyRef) => {
-          if (expPhysicalName == "UUID") {
-            assertThat(k).asString.startsWith("col-")
-          } else {
-            assertThat(k).asString.isEqualTo(expPhysicalName)
-          }
-        })
+      assertColumnMapping(updatedMetadata.getSchema.get("a"), 1L)
+
+      // Now add few more fields to the same schema
+      val updateSchema = updatedMetadata.getSchema
+        .add("b", StringType.STRING)
+        .add("c", new ArrayType(IntegerType.INTEGER, false))
+        .add("d", new MapType(IntegerType.INTEGER, StringType.STRING, false))
+        .add("e", new StructType().add("h", IntegerType.INTEGER))
+
+      val inputMetadata2 = testMetadata(updateSchema)
+      val updatedMetadata2 = updateColumnMappingMetadataIfNeeded(
+        if (icebergCompatV2Enabled) inputMetadata2.withIcebergCompatV2Enabled else inputMetadata2,
+        ID,
+        false).get()
```

Comment:
nit - instead of these unchecked gets we probably should assert the result is non-empty (maybe helper fx `getOrElse`?)